plotter/scatter.go

```diff
@@ -24,22 +28,32 @@ type Scatter struct {
 // NewScatter returns a Scatter that uses the
 // default glyph style.
 func NewScatter(xys XYer) (*Scatter, error) {
+	var s Scatter
 	data, err := CopyXYs(xys)
 	if err != nil {
 		return nil, err
 	}
-	return &Scatter{
-		XYs:        data,
-		GlyphStyle: DefaultGlyphStyle,
-	}, err
+	s.XYs = data
+
+	s.GlyphStyle = DefaultGlyphStyle
+
+	s.GlyphStyleFunc = func(int) draw.GlyphStyle {
```

Comment:
we could keep it `nil`, couldn't we? (as we handle the `nil` case in the `Scatter.Plot` method)

actually, I think we could leave the whole `NewScatter` function untouched.

Reply:
my comment above is still relevant, I believe.

Reply:
When I make it `nil`, I got a warning about a unability to use it in this case.

And a lot of tests (timeseries, for instance) use `DeafultGlyphStyle`, so a lot of tests fail in this case.

Reply:
I am talking about `GlyphStyleFunc` being `nil`. not `GlyphStyle` (which is indeed set by default to `DefaultGlyphStyle`.)

I believe that, if you leave the whole `NewScatter` unmodified with regard to what it was before, then everything should work well.

ie: `NewScatter` should look like:
```go
// NewScatter returns a Scatter that uses the
// default glyph style.
func NewScatter(xys XYer) (*Scatter, error) {
	data, err := CopyXYs(xys)
	if err != nil {
		return nil, err
	}
	return &Scatter{
		XYs:        data,
		GlyphStyle: DefaultGlyphStyle,
	}, err
}
```

Reply:
Ooops, sorry. Did not understand that. Now it works with the original `NewScatter`.